src/common/reducers/entities.js

```diff
@@ -0,0 +1,66 @@
+import merge from 'lodash/merge';
+
+import * as ActionTypes from '../actions/repository';
+
+// export only for testing
+export function repository(state={
+  __isSelected: false,
+  __isFetching: false,
+  __error: null
+}, action) {
+
+  switch(action.type) {
```

Comment:
I don't fully understand why a reducer for the selected repository appears in here and not in a separate module.